engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingStateMachine.java

```diff
@@ -260,7 +262,15 @@ private void processCommand(final LoggedEvent command) {
           () -> {
             processingResultBuilder.reset();
 
-            currentProcessingResult = engine.process(typedCommand, processingResultBuilder);
+            currentProcessor =
+                recordProcessors.stream()
+                    .filter(p -> p.canProcess(typedCommand.getValueType()))
+                    .findFirst()
+                    .orElse(null);
+            if (currentProcessor != null) {
+              currentProcessingResult =
+                  currentProcessor.process(typedCommand, processingResultBuilder);
+            }
```

Comment:
:wrench: I think I would do that outside of the transaction. So first try to choose the right processor, if we found one then start the processing (including start metrics and transaction)